api/http/status.go

```diff
@@ -64,3 +66,18 @@ func (h StatusController) Health(c *gin.Context) {
 
 	c.Writer.WriteHeader(http.StatusNoContent)
 }
+
+// Shutdown responds to GET /shutdown
+func (h StatusController) Shutdown(c *gin.Context) {
+	pid := os.Getpid()
+	err := syscall.Kill(pid, syscall.SIGUSR1)
+	if err != nil {
+		ctx := c.Request.Context()
+		log.FromContext(ctx).Error(errors.Wrap(err, "shutdown failed"))
+		c.JSON(http.StatusServiceUnavailable, gin.H{
+			"error": err.Error(),
+		})
+		return
+	}
+	c.Writer.WriteHeader(http.StatusAccepted)
```

Comment:
I just realized that if this should work with a preStop hook, the handler needs to hang until the application shuts down, otherwise it will simply receive a SIGTERM immediately after.

Reply:
Done :)